src/dswx_sar/region_growing.py

```diff
@@ -0,0 +1,308 @@
+import time
+import os
+import numpy as np
+from scipy import ndimage
+import cv2
+
+import time
+import logging
+import mimetypes
+from joblib import Parallel, delayed
+
+from dswx_sar import dswx_sar_util
+from dswx_sar.dswx_runconfig import RunConfig, _get_parser
+
+
+logger = logging.getLogger('dswx_s1')
+
+
+def region_growing(fuzz_image,
+                   initial_seed=0.6,
+                   tolerance=0.45,
+                   maxiter=200):
+    """The regions are then grown from the seed points to adjacent points
+    since it covers the tolerance values.
+
+    Parameters
+    ----------
+    fuzz_image : numpy.ndarray
+        fuzzy image with values [0, 1] where 0 is non-water and 1 is water
+    initial_seed : float
+        Threshold used for the seed of water body [0 ~ 1]
```

Comment:
Your description is more specific and better for this specific purpose. But in my opinion, including 'water' in the variable name may be too specific. 

The reason why the initial threshold is always higher than tolerance is that the region growing starts from the initial threshold (seed) and classify the fuzzy image with the initial threshold to initially find the high likelihood of water and remove the wrong classified pixels. 

After that, in the next loop, the code looks for adjacent pixels within the relaxed values to find more water areas. This is because some pixels in water objects can have slightly lower fuzzy values due to the windy conditions.

Reply:
Thank for the explanation.

What do you think of either `seed_threshold` or `initial_threshold`? Without looking at the docstring I think either suggestion is more specific.